File: Admission.Api/Models/Application.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Admission.Api.Models
{
    public class Application
    {
        public int Id { get; set; }

        [Required]
        public int ApplicantId { get; set; }
        public Applicant? Applicant { get; set; }

        [Required, MaxLength(255)]
        public string Faculty { get; set; } = null!;

        [Required, MaxLength(255)]
        public string Specialty { get; set; } = null!;

        [MaxLength(255)]
        public string? EducationalInstitution { get; set; }

        public int? GraduationYear { get; set; }

        [MaxLength(100)]
        public string? DocumentType { get; set; }

        [MaxLength(50)]
        public string? DocumentNumber { get; set; }

        [Column(TypeName = "numeric(5,2)")]
        public decimal? AverageScore { get; set; }

        [Column(TypeName = "numeric(5,2)")]
        public decimal? EgeScore { get; set; }

        [MaxLength(10)]
        public string? GroupNumber { get; set; }

        public ICollection<SubjectGrade> SubjectGrades { get; set; } = new List<SubjectGrade>();
        public ICollection<ExamResult> ExamResults { get; set; } = new List<ExamResult>();
    }
}

File: admissionapi_front/lib/models/application.dart
class Application {
  int? id;
  int applicantId;
  String faculty;
  String specialty;
  String? educationalInstitution;
  int? graduationYear;
  String? documentType;
  String? documentNumber;
  double? averageScore;
  double? egeScore;
  String? groupNumber;

  Application({
    this.id,
    required this.applicantId,
    required this.faculty,
    required this.specialty,
    this.educationalInstitution,
    this.graduationYear,
    this.documentType,
    this.documentNumber,
    this.averageScore,
    this.egeScore,
    this.groupNumber,
  });

  factory Application.fromJson(Map<String, dynamic> json) {
    return Application(
      id: json['Id'] ?? json['id'],
      applicantId: json['ApplicantId'] ?? json['applicantId'] ?? 0,
      faculty: (json['Faculty'] ?? json['faculty'] ?? '').toString(),
      specialty: (json['Specialty'] ?? json['specialty'] ?? '').toString(),
      educationalInstitution: json['EducationalInstitution'] ?? json['educationalInstitution'],
      graduationYear: json['GraduationYear'] ?? json['graduationYear'],
      documentType: json['DocumentType'] ?? json['documentType'],
      documentNumber: json['DocumentNumber'] ?? json['documentNumber'],
      averageScore: (json['AverageScore'] ?? json['averageScore'])?.toDouble(),
      egeScore: (json['EgeScore'] ?? json['egeScore'])?.toDouble(),
      groupNumber: json['GroupNumber'] ?? json['groupNumber'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'applicantId': applicantId,
      'faculty': faculty,
      'specialty': specialty,
      'educationalInstitution': educationalInstitution,
      'graduationYear': graduationYear,
      'documentType': documentType,
      'documentNumber': documentNumber,
      'averageScore': averageScore,
      'egeScore': egeScore,
      'groupNumber': groupNumber,
    };
  }
}